### DISCUSSION TITLE
ERR_ERL_DOUBLE_COUNT with multiple rate limits

### DISCUSSION THREAD

**Discussion by stephent** [nfriedly/express-rate-limit#393] [Ideas]
Before this warning was added, I'd had multiple rate limits configured for a long time, all pointing to the same Redis store. With the addition of the warning, the double counting is now apparent.

However, wouldn't it be a friendlier approach to have the library look after unique scoping keys per configured rate limit rather than imposing this on the library user? If you're able to configure multiple rate limits, I suspect many would naturally assume this would already be the case by default, but it seems not.

(Not sure if there are other use cases which make this difficult or undesirable to implement?)

UDPATE: on reviewing my code more thoroughly, I see I already have unique `prefix` values per rate limit for the RedisStore. Confused as to why I'm seeing this error...

**Comment by nfriedly** [comment]
Oof, I think that's a bug in the validation check. I would recommend that you said `validate: false` in your express-rate-limit configuration for right now, and I'll try and figure out a fix.

**Comment by stephent** [comment]
Thanks @nfriedly - I removed the prefix and added custom keyGenerator functions and all seems well again now.
